index.js

```diff
@@ -87,54 +86,64 @@ cm.Item({
     cm.URLContext(['*.youtube.com']),
     cm.SelectorContext('[class*="yt-uix-sessionlink"]')
   ],
-  contentScript: 'self.on("click", function (node, data) {' +
-                 '  self.postMessage(node.href);' +
-                 '});',
-  onMessage: function(url) {
-    const id = getVideoId(url);
-    updatePanel({domain: 'youtube.com', id: id, src: ''});
-    getYouTubeUrl(id, function(err, streamUrl) {
-      if (!err) updatePanel({src: streamUrl});
-    });
-  }
+  contentScript: contextMenuContentScript,
+  onMessage: launchYouTube
 });
 
 cm.Item({
   label: 'Send to mini player',
   context: cm.SelectorContext('[href*="vimeo.com"]'),
-  contentScript: 'self.on("click", function (node, data) {' +
-                 '  self.postMessage(node.href);' +
-                 '});',
-  onMessage: function(url) {
-
-    const id = getVideoId(url);
-    updatePanel({domain: 'vimeo.com', id: id, src: ''});
-    getVimeoUrl(id, function(err, streamUrl) {
-      if (!err) updatePanel({src: streamUrl});
-    });
-  }
+  contentScript: contextMenuContentScript,
+  onMessage: launchVimeo
 });
 
 function updatePanel(opts) {
   panel.port.emit('set-video', opts);
   panel.show();
 }
 
-const pageMod = require('sdk/page-mod');
+function launchYouTube(url) {
+  const id = getVideoId(url);
+  updatePanel({domain: 'youtube.com', id: id, src: ''});
+  getYouTubeUrl(id, function(err, streamUrl) {
+    if (!err) updatePanel({src: streamUrl});
+  });
+}
 
+function launchVimeo(url) {
+  const id = getVideoId(url);
+  updatePanel({domain: 'vimeo.com', id: id, src: ''});
+  getVimeoUrl(id, function(err, streamUrl) {
+    if (!err) updatePanel({src: streamUrl});
+  });
+}
```

Comment:
As we add more video providers, it might be nice to just have one function and pass in the stuff that varies as a data object:
`{ url: url, getUrlFn: getVimeoUrl, domain: 'vimeo.com' }`
or
`{ url: url, getUrlFn: getVideoId, domain: 'youtube.com' }`